### PR TITLE
Implement build async

### PR DESCRIPTION
**Pull request by itsmontoya**

### CODE REVIEW

File: plugins.go [hunk 1]
```diff
@@ -106,6 +107,27 @@ func (p *Plugins) Build() (err error) {
...
+func (p *Plugins) BuildAsync(q *queue.Queue) (err error) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	var wg sync.WaitGroup
+	wg.Add(len(p.ps))
+
+	var errs errors.ErrorList
+	for _, pi := range p.ps {
+		q.New(func() {
```
Comment (cephalization) [inline plugins.go:120]: Does `q.New` call the anonymous function in a go-routine?

Comment (itsmontoya) [reply plugins.go:120]: `q.New` pushes the func to the queue :)

**Review by dhalman — APPROVED** [review]

File: plugins.go (hunk 1 above)
Comment (cephalization) [reply plugins.go:120]: If it is in a queue, what makes it asynchronous? Does the queue trigger builds in different goroutines?

Comment (itsmontoya) [reply plugins.go:120]: the queue has x number of goroutines listening to it. So it allows you to easily spread tasks over x threads. Check out the queue library :) https://github.com/hatchify/queue

**Pull request merged**